src/main/java/com/example/travelservice/services/TripServiceImpl.java

```diff
@@ -44,4 +44,15 @@ public List<WeatherDto> showTripWeather(Trip trip) {
 
     }
 
+    @Override
+    public Trip updateTrip(Long id, Trip trip) {
+
+        trip.setId(id);
+
+        if (tripRepository.findById(id).isPresent()) {
+            return tripRepository.save(trip);
+        } else {
+            return createTrip(trip);
```

Comment:
same here, do not create new entity if user is trying to update non-existing trip, it's not correct behaviour

Reply:
logic changed